### ISSUE TITLE
It doesn't look like `assets/misc/neon_nmdc_term_mapping.tsv` is being maintained with respect to schema changes

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/nmdc-schema#1703]
[assets/misc/neon_nmdc_term_mapping.tsv](https://github.com/microbiomedata/nmdc-schema/blob/86b26440311d01b2850aa0c7210930e13dc17531/assets/misc/neon_nmdc_term_mapping.tsv#L314) is still mentioning `quality_control_report.status`, which will be removed in the pending release

We shouldn't have any artifacts that mention patterns that are illegal in the released schema

- What is the role of `assets/misc/neon_nmdc_term_mapping.tsv`? 
- Is it being programmatically accessed?
- Are there any related files that might be getting stale in the same way?

**Comment by turbomam** [comment]
The mapping files should be tested!

**Comment by turbomam** [comment]
`depth` mapping is **interesting** too

`depth` source would ideally be `lakeSampleDepth1` etc. but there is a lots of inter-field dependency here

TLDR flat file doesn't really capture the mapping!

**Comment by turbomam** [comment]
see also 
- #1719

**Comment by turbomam** [comment]
long term: could use [linkml-transformer](https://github.com/linkml/linkml-transformer)
